Tournament.cpp

```diff
@@ -0,0 +1,221 @@
+#include "Tournament.h"
+
+	Tournament::Tournament()
+	{
+		std::string EuroPath = "../FIFApoints/EuroCupParticipants.txt";
+		std::string EuroGroupPath = "../FIFApoints/EuroCupGroups.txt";
+		std::string WorldCupPath = "../FIFApoints/WorldCupParticipants.txt";
+		std::string WorldCupGroupPath = "../FIFApoints/WorldCupGroups.txt";
+		switch (ChooseTournament())
+		{
+		case 1:
+			AddParticipants(EuroPath, m_euroParticipants);
+			SetGroup(EuroGroupPath, m_numberEuroGroup);
+			break;
+		case 2:
+			AddParticipants(WorldCupPath, m_worldCupParticipants);
+			SetGroup(WorldCupGroupPath, m_numberWorldCupGroup);
+			break;
+		default:
+			std::cout << "Default case!" << std::endl;
+			break;
+		}
+	}
+	Tournament::~Tournament() {}
+	int Tournament::ChooseTournament()
+	{
+		int choice;
+		do {
+			std::cout << "---Select a tournament---" << std::endl;
+			std::cout << "Press 1 for Euro or 2 for World Cup!" << std::endl;
+			std::cin >> choice;
+		} while (choice < 1 or choice > 2);
+		return choice;
+	}
+
+	void Tournament::AddParticipants(std::string path, int numberParticipants)
+	{
+		std::ifstream ParticipantsList;
+		ParticipantsList.open(path);
+		if (!ParticipantsList.is_open() and !ParticipantsList.good())
+			std::cout << "File is not open!" << std::endl;
+		else
+		{
+			std::string name;
+			int rank;
+			double fifa_points;
+			for (int i = 0; i < numberParticipants; ++i)
+			{
+				ParticipantsList >> name >> rank >> fifa_points;
+				std::shared_ptr<Team> p_new_team(new Team(name, rank, fifa_points));
+				m_participants.push_back(p_new_team);
+			}
+		}
+		ParticipantsList.close();
+	}
+	void Tournament::SetGroup(std::string path, const int tournamentGroups)
+	{
+		std::ifstream GroupList;
+		GroupList.open(path);
+		if(!GroupList.is_open() and !GroupList.good())
+			std::cout << "File is not open!" << std::endl;
+		else
+		{
+			std::string name;
```

Comment:
Don't use tabs, use 2 or 4 spaces, it can be easily configured in any editor. Otherwise code with high indentation levels quickly becomes unreadable.